scrapy/downloadermiddlewares/httpcache.py

```diff
@@ -73,7 +88,7 @@ def process_request(self, request, spider):
         # process_response hook
         request.meta['cached_response'] = cachedresponse
 
-    def process_response(self, request, response, spider):
+    def process_response(self, request: Type[Request], response: Type[Response], spider: Type[Spider]) -> Type[Response]:
```

Comment:
Type is not needed here, and in other places as well. Type means that a class instance is received, not an object of this type, which is not the case.

Reply:
Right, I find the right way is to use TypeVar.

Reply:
TypeVar is removed. Now the native class is adopted for typing hint.